src/extension/src/coreTemplateStudio.ts

```diff
@@ -94,144 +84,143 @@ export class CoreTemplateStudio {
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.PROJECT_TYPE,
-      projectType
-    );
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.FRONTEND_FRAMEWORK,
-      frontendFramework
-    );
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.BACKEND_FRAMEWORK,
-      backendFramework
-    );
-    return await fetch(url.href, { method: CONSTANTS.API.METHODS.GET })
-      .then((response: Response) => {
-        return response.json();
-      })
-      .catch((error: Error) => {
-        throw Error(error.toString());
-      });
+    const getPagesCommand = `${CONSTANTS.CLI.GET_PAGES_COMMAND_PREFIX} -p ${projectType} -f ${frontendFramework} -b ${backendFramework}\n`;
+    this._processCli.stdin.write(getPagesCommand);
+    return await this.awaitCliEvent(CONSTANTS.CLI.GET_PAGES_COMPLETE_STATE);
   }
 
-  public async getPages(
+  public async getFeatures(
     projectType: string,
     frontendFramework: string,
     backendFramework: string
   ): Promise<any> {
-    // TODO: use this in client instead of fetching directly from API server
-    const url = new URL(CONSTANTS.API.ENDPOINTS.PAGE, this._url);
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.PROJECT_TYPE,
-      projectType
-    );
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.FRONTEND_FRAMEWORK,
-      frontendFramework
-    );
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.BACKEND_FRAMEWORK,
-      backendFramework
-    );
-
-    return await fetch(url.href, { method: CONSTANTS.API.METHODS.GET })
-      .then((response: Response) => {
-        return response.json();
-      })
-      .catch((error: Error) => {
-        throw Error(error.toString());
-      });
+    // to use this in client
+    const getFeaturesCommand = `${CONSTANTS.CLI.GET_FEATURES_COMMAND_PREFIX} -p ${projectType} -f ${frontendFramework} -b ${backendFramework}\n`;
+    this._processCli.stdin.write(getFeaturesCommand);
+    return await this.awaitCliEvent(CONSTANTS.CLI.GET_FEATURES_COMPLETE_STATE);
   }
 
-  public async sync(payload: ICommandPayload): Promise<any> {
-    let connection = await this.connectToCoreApiHub();
-    let command = new SyncCommand(payload);
-    return await command.execute(connection);
+  public async getProjectTypes(): Promise<any> {
+    // to use this in client
+    const getProjectTypesCommand = `${CONSTANTS.CLI.GET_PROJECT_TYPES_COMMAND_PREFIX}\n`;
+    this._processCli.stdin.write(getProjectTypesCommand);
+    return await this.awaitCliEvent(CONSTANTS.CLI.GET_PROJECT_TYPES_COMPLETE_STATE);
   }
 
   public async generate(payload: ICommandPayload): Promise<any> {
-    let connection = await this.connectToCoreApiHub();
-    let command = new GenerateCommand(payload);
-    return await command.execute(connection);
+    var generatePayload = JSON.stringify(this.makeEngineGenerationPayload(<IGenerationPayloadType>(payload.payload)));
```

Comment:
Should be const as we don't modify the variable

Reply:
fixed in my PR :)